text/0000-get-collection-keys.md

```diff
@@ -0,0 +1,121 @@
+- Feature Name: get_collection_keys
+- Start Date: 2015-06-24
+- RFC PR: https://github.com/rust-lang/rust/pull/26531
+- Rust Issue:
+
+# Summary
+
+Get the original keys back from collections.
+
+# Motivation
+
+Removing a key from a HashMap or an item from a HashSet will drop the item.
+This is usually acceptable for small clonable objects like numbers and strings,
+but they are not always clonable, and even if they are it might be an expensive
+operation.
+
+# Detailed design
+
+There are currently four operations that could have a new function to them:
+`get`, `get_mut`, `insert` and `remove`. All of those methods receive a reference
+to a key type and return reference to the value or a the value itself.
```

Comment:
Also this sentence is troubled. Might I suggest "None of these methods provide access to the key" or something.